tests/object/tree/read.c

```diff
@@ -73,3 +73,59 @@ void test_object_tree_read__two(void)
 	git_object_free(obj);
 	git_tree_free(tree);
 }
+
+#define BIGFILE "bigfile"
+
+void test_object_tree_read__largefile(void)
+{
+	git_reference *ref;
+	git_commit *commit;
+	git_tree *tree;
+	git_oid oid;
+	const git_tree_entry *entry;
+	git_object *object;
+	git_buf file = GIT_BUF_INIT;
+	int fd;
+	fstore_t prealloc;
+	git_index *idx;
+
+	if (!cl_is_env_set("GITTEST_INVASIVE_FS_SIZE"))
+		cl_skip();
+
+	cl_git_pass(git_reference_lookup(&ref, g_repo, "refs/heads/master"));
+	cl_git_pass(git_repository_index(&idx, g_repo));
+
+	git_buf_puts(&file, git_repository_workdir(g_repo));
+	git_buf_joinpath(&file, file.ptr, BIGFILE);
+	cl_git_pass(git_buf_oom(&file));
+
+	fd = p_open(git_buf_cstr(&file), O_CREAT|O_RDWR, 0644);
+	cl_assert_(fd >= 0, "invalid file descriptor");
+
+	memset(&prealloc, 0, sizeof(prealloc));
+	prealloc.fst_flags  = F_ALLOCATEALL;
+	prealloc.fst_posmode = F_PEOFPOSMODE;
+	prealloc.fst_offset = 0;
+	prealloc.fst_length = (off_t)4 * 1024 * 1024 * 1024; /* 4 GiB */
```

Comment:
Unfortunately, this is not portable at all. But there's `posix_fallocate` -- it requires _POSIX_C_SOURCE>=200112L, which is currently only set on SunOS and Solaris. No idea why.

Reply:
As macOS doesn't have that, I'm not sure how to proceed (or if it's worth implementing a shim for a "successful" test case).

Reply:
I've used preprocessor tricks to make it work on Linux/macOS.